test/unit/request-auth.test.js

```diff
@@ -4,123 +4,185 @@ var expect = require('chai').expect,
+        it('should be properly initialised when set from Request constructor in legacy format', function () {
+            var request = new Request({
+                auth: {
+                    noauth: { foo: 'bar' },
+                    type: 'noauth'
+                }
+            });
+
+            expect(request.auth.current()).to.eql({ foo: 'bar' });
+        });
     });
 
-    it('should be properly initialised when set from Request constructor in legacy format', function () {
-        var request = new Request({
-            auth: {
+    describe('.use', function () {
+        it('should be able to set a new auth using .use', function () {
+            var auth = new RequestAuth({
                 noauth: { foo: 'bar' },
                 type: 'noauth'
-            }
-        });
+            });
 
-        expect(request.auth.current()).to.eql({ foo: 'bar' });
-    });
+            expect(auth.current()).to.eql({ foo: 'bar' });
 
-    it('should be able to set a new auth using .use', function () {
-        var auth = new RequestAuth({
-            noauth: { foo: 'bar' },
-            type: 'noauth'
+            auth.use('basic');
+            expect(auth.type).to.equal('basic');
+            expect(auth.basic.toJSON()).to.eql([]);
+            expect(auth.current()).to.eql({});
         });
 
-        expect(auth.current()).to.eql({ foo: 'bar' });
-
-        auth.use('basic');
-        expect(auth.type).to.equal('basic');
-        expect(auth.basic.toJSON()).to.eql([]);
-        expect(auth.current()).to.eql({});
-    });
+        it('should be able to construct with multiple auths', function () {
+            var auth = new RequestAuth({
+                noauth: {
+                    foo: 'bar'
+                },
+                basic: {
+                    username: 'u', password: 'p'
+                },
+                type: 'noauth'
+            });
 
-    it('should be able to construct with multiple auths', function () {
-        var auth = new RequestAuth({
-            noauth: {
+            expect(auth.current()).to.eql({
                 foo: 'bar'
-            },
-            basic: {
+            });
+
+            auth.use('basic');
+            expect(auth.type).to.equal('basic');
+            expect(auth.current()).to.eql({
                 username: 'u', password: 'p'
-            },
-            type: 'noauth'
+            });
         });
 
-        expect(auth.current()).to.eql({
-            foo: 'bar'
-        });
+        it('should be able to return the parameters of the selected auth type', function () {
+            var auth = new RequestAuth({
+                noauth: {
+                    foo: 'bar'
+                },
+                type: 'noauth'
+            });
 
-        auth.use('basic');
-        expect(auth.type).to.equal('basic');
-        expect(auth.current()).to.eql({
-            username: 'u', password: 'p'
+            auth.use('noauth', {
```

Comment:
Why are we setting the auth options again in the `.use` method? The test name does not say anything about it.
It is making the `expect` a little confusing.

If we're testing that `.use` can update the options, I think we can break this test into two.

Reply:
It's the latter one, have updated the test description to depict that we are updating the auth via options.

Reply:
I think it's better for readability that we pass just `{ foo1: 'bar1' }` rather than the whole thing again.
It also makes the usage of the function more clear.